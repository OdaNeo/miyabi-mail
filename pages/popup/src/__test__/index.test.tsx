import { describe, it, expect, beforeEach, vi } from 'vitest';

vi.mock('@/Popup', () => ({
  default: () => <div data-testid="popup">Mock Popup Component</div>,
}));

vi.mock('@/index.css', () => ({}));

describe('Popup initialization', () => {
  beforeEach(() => {
    document.body.innerHTML = '';
    vi.resetModules();
    vi.clearAllMocks();
  });

  it('should render Popup correctly', async () => {
    document.body.innerHTML = '<div id="app-container"></div>';
    const { init } = await import('../index');
    init();
    const container = document.querySelector('#app-container');
    expect(container).toMatchSnapshot();
  });

  it('should throw error when container is missing', async () => {
    try {
      await import('../index');
      throw new Error('Expected error was not thrown');
    } catch (error: any) {
      expect(error.message).toContain('Can not find #app-container');
    }
  });
});
