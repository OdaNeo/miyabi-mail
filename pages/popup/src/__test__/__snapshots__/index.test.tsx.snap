// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popup initialization > should render Popup correctly 1`] = `
<div
  id="app-container"
/>
`;
